Zed reno changes

Alex-Welsh opened the pull request:
This PR contains various commits that relate to upstream changes for Zed, from reading through the release notes

Alex-Welsh committed:
Update selinux reboot variable name

Required by change I42933b0b7d55c69c9f6992e331fafb2e6c42d4d1

Alex-Welsh committed:
Add template config for swap configuration

Alex-Welsh committed:
Remove references to kolla_install_type

Alex-Welsh committed:
Remove references to Monasca for Zed release

Alex-Welsh committed:
Remove references to Elasticsearch

Elasticsearch is no longer supported these chages remove references to
Elasticsearch and replaces them with opensearch where necessary

Alex-Welsh committed:
Remove definitions for neutron/glance_tls_proxy

In Zed, the default value for neutron_tls_proxy and glance_tls_proxy is
haproxy_tag, making the explicit definition redundant.

markgoddard commented on .github/workflows/stackhpc-container-image-build.yml:
```diff
@@ -198,6 +198,9 @@ jobs:
 
+      # NOTE(Alex-Welsh) This may not work after the zed release, since the
+      # kolla_version tag is changing to be a sha-1 hash if images are built
+      # with kolla from a git repository
```
I think this just filters images with a label of kolla_version, so should still work

Alex-Welsh replied:
Removed

markgoddard commented on doc/source/configuration/wazuh.rst:
```diff
@@ -73,7 +73,7 @@ Optional: define LVM volumes ``etc/kayobe/inventory/group_vars/wazuh-manager/lvm
           filesystem: "ext4"
           mount: true
-          mntp: “/var/lib/elasticsearch”
+          mntp: “/var/lib/opensearch”
```
Unsure about this one. Does wazuh actually use this directory for OpenSearch? If so this should probably go in yoga too.

Alex-Welsh replied:
I think so but I'm also not sure. Might be the sort of thing we just need to shake out during multinode testing.

markgoddard replied:
I think let's stick with elasticsearch for now since the move to Zed isn't changing anything to do with wazuh.

Alex-Welsh replied:
Done

markgoddard commented on etc/kayobe/environments/aufn-ceph/kolla/globals.yml:
```diff
@@ -12,5 +12,5 @@ glance_backend_ceph: "yes"
 
-# Elasticsearch memory tuning
+# Opensearch memory tuning
 es_heap_size: 1g
```
```suggestion
opensearch_heap_size: 1g
```

Alex-Welsh replied:
Done

markgoddard commented on etc/kayobe/environments/ci-aio/kolla/globals.yml:
```diff
@@ -12,7 +12,7 @@ openstack_service_rpc_workers: "1"
 
-# Elasticsearch memory tuning
+# Opensearch memory tuning
 es_heap_size: 1g
```
```suggestion
opensearch_heap_size: 1g
```

Alex-Welsh replied:
Done

markgoddard commented on etc/kayobe/environments/ci-multinode/kolla/globals.yml:
```diff
@@ -15,7 +15,7 @@ nova_backend_ceph: "yes"
 
-# Elasticsearch memory tuning
+# Opensearch memory tuning
 es_heap_size: 1g
```
```suggestion
opensearch_heap_size: 1g
```

Alex-Welsh replied:
Done

markgoddard commented on etc/kayobe/compute.yml:
```diff
@@ -196,6 +196,13 @@
+
+# List of swap devices. Each item is a dict containing a 'device' item. Default
+# is an empty list.
+# compute_swap:
```
Looks like these didn't make it into etc/kayobe/compute.yml in kayobe

Alex-Welsh replied:
Interesting. Do we need to change anything here or it an Upstream PR?

markgoddard replied:
Nothing to change in here, but an upstream patch would be nice.

markgoddard commented on etc/kayobe/ironic.yml:
```diff
@@ -97,14 +97,14 @@
-# List of default kernel parameters to append for baremetal PXE boot.
-#kolla_ironic_pxe_append_params_default:
+# List of default kernel parameters to append for baremetal PXE (TFTP) boot.
+#kolla_ironic_tftp_append_params_default:
```
Not seeing these in Kayobe stable/zed branch

Alex-Welsh replied:
I think these were a response to [this change](https://review.opendev.org/c/openstack/kolla-ansible/+/832134). I just assumed it had all propagated through Kayobe too.

Alex-Welsh replied:
Will remove them in next patchset

Alex-Welsh replied:
Done

markgoddard commented on etc/kayobe/kolla.yml:
```diff
@@ -26,10 +26,6 @@ kolla_source_version: "{{ stackhpc_kolla_source_version }}"
 
-# Type of Kolla-ansible control installation. One of 'binary' or 'source'.
-# Default is 'source'.
-#kolla_ansible_ctl_install_type:
```
This is still relevant, it's about the kolla-ansible python package installation

Alex-Welsh replied:
Whoops. Reverted

markgoddard commented on etc/kayobe/kolla/globals.yml:
```diff
@@ -1,6 +1,10 @@
 
+# TODO(Alex-Welsh): Check whether this is necessary and this is the correct
+# place to put it
+#image_name_prefix: "{{ os_distribution }}"
```
Not used - remove?

Alex-Welsh replied:
Whoops. Reverted

markgoddard commented on etc/kayobe/pulp.yml:
```diff
@@ -1172,7 +1143,7 @@ stackhpc_pulp_repository_container_repos_kolla: >-
   {%- for image in stackhpc_pulp_images_kolla_filtered -%}
   {%- if image not in stackhpc_kolla_unbuildable_images[kolla_base_distro]-%}
-  {%- set image_repo = kolla_docker_namespace ~ "/" ~ kolla_base_distro ~ "-" ~ kolla_install_type ~ "-" ~ image -%}
+  {%- set image_repo = kolla_docker_namespace ~ "/" ~ kolla_base_distro ~ "-" ~ image -%}
```
Also remove kolla_base_distro?

Alex-Welsh replied:
Removed

markgoddard commented on etc/kayobe/pulp.yml:
```diff
@@ -1189,7 +1160,7 @@ stackhpc_pulp_distribution_container_kolla: >-
   {%- for image in stackhpc_pulp_images_kolla_filtered -%}
   {%- if image not in stackhpc_kolla_unbuildable_images[kolla_base_distro]-%}
-  {%- set image_repo = kolla_docker_namespace ~ "/" ~ kolla_base_distro ~ "-" ~ kolla_install_type ~ "-" ~ image -%}
+  {%- set image_repo = kolla_docker_namespace ~ "/" ~ kolla_base_distro ~ "-" ~ image -%}
```
ditto

Alex-Welsh replied:
Done

Alex-Welsh committed:
Zed configuration changes

This commit includes two changes.
The first is a change to  requirements.yml to no longer consider python
versions < 3.8
The second updates references to ``ironic_pxe`` to ``ironic_tftp``.

Alex-Welsh committed:
Fix Zed-specific changes

This commit includes two fixes.
The first swaps instances of es_heap_size with opensearch_heap_size.
The second removes kolla_base_distro from name of containers in
the local pulp repository.
Reverted changes to docs for wazuh/opensearch integration.

Alex-Welsh commented:
Going to need you to force merge this unfortunately @markgoddard. CI's still blocking
